src/__experimental__/components/checkbox/checkbox.style.js

```diff
@@ -153,11 +188,49 @@ CheckboxStyle.defaultProps = {
 CheckboxStyle.propTypes = {
   disabled: PropTypes.bool,
   error: PropTypes.bool,
+  hasError: PropTypes.bool,
   fieldHelpInline: PropTypes.bool,
   inputWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
   labelAlign: PropTypes.string,
   labelWidth: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
   size: PropTypes.string
 };
 
+const StyledCheckboxGroup = styled.div`
+  ${({
+    theme,
+    hasError,
+    hasWarning,
+    hasInfo
+  }) => css`
+    & ${CheckboxStyle} {
```

Comment:
Are these necessary? If you pass the correct prop to one of the Checkboxes, they will already have the relevant styles because they're defined in `CheckboxStyle` above.